diff --git i/Gemfile w/Gemfile
index ace0891..7613be8 100644
--- i/Gemfile
+++ w/Gemfile
@@ -41,6 +41,8 @@ gem 'bootsnap', '>= 1.1.0', require: false
 gem 'foundation-rails', '~> 6.0' # required
 gem 'foundation_rails_helper', '~> 3.0'
 
+gem 'jquery-rails'
+
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
diff --git i/Gemfile.lock w/Gemfile.lock
index 820f524..ce70070 100644
--- i/Gemfile.lock
+++ w/Gemfile.lock
@@ -113,6 +113,10 @@ GEM
     jbuilder (2.7.0)
       activesupport (>= 4.2.0)
       multi_json (>= 1.2)
+    jquery-rails (4.3.3)
+      rails-dom-testing (>= 1, < 3)
+      railties (>= 4.2.0)
+      thor (>= 0.14, < 2.0)
     json (2.1.0)
     listen (3.1.5)
       rb-fsevent (~> 0.9, >= 0.9.4)
@@ -299,6 +303,7 @@ DEPENDENCIES
   guard-rspec (~> 4.7.3)
   guard-rubocop (~> 1.3.0)
   jbuilder (~> 2.5)
+  jquery-rails
   listen (>= 3.0.5, < 3.2)
   pronto
   pronto-rubocop (~> 0.9.1)
diff --git i/app/assets/javascripts/application.js w/app/assets/javascripts/application.js
index 78d462f..81c5edd 100644
--- i/app/assets/javascripts/application.js
+++ w/app/assets/javascripts/application.js
@@ -10,6 +10,7 @@
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
 //
+//= require jquery
 //= require rails-ujs
 //= require foundation
 //= require activestorage
@@ -17,3 +18,37 @@
 //= require_tree .
 
 $(function(){ $(document).foundation(); });
+
+$(function(){
+  $('#query').on('keyup', function(event){
+    var input = $(event.target)
+    var query = input.val()
+
+    $.ajax({
+      method: "GET",
+      url: "/tasks/search",
+      data: { query: query }
+    })
+    .done(function( suggestions ) {
+      var $suggestions = $('#suggestions')
+      var $suggestions_table = $suggestions.find('table tbody')
+
+      $suggestions_table.empty()
+
+      suggestions.forEach(function(suggestion) {
+        $suggestions_table.append('<tr><td>' + suggestion + '</td></tr>')
+      });
+
+      $suggestions.show()
+    });
+  })
+
+  $('#suggestions table').on('click', function(event){
+      $('#suggestions').hide()
+
+      var row = $(event.target)
+
+      $('#query').val(row.html())
+      $('#query').closest('form').submit()
+  })
+})
diff --git i/app/assets/stylesheets/application.css w/app/assets/stylesheets/application.css
index a152282..112cbea 100644
--- i/app/assets/stylesheets/application.css
+++ w/app/assets/stylesheets/application.css
@@ -15,3 +15,21 @@
  *= require foundation_and_overrides
 
  */
+
+
+#suggestions {
+	display: none;
+	position: relative;
+}
+
+#suggestions table {
+	position: absolute;
+	border: 2px solid;
+
+	-webkit-box-shadow: 0px 0px 50px 0px rgba(0,0,0,0.75);
+	-moz-box-shadow: 0px 0px 50px 0px rgba(0,0,0,0.75);
+	box-shadow: 0px 0px 50px 0px rgba(0,0,0,0.75);
+}
+
+
+
diff --git i/app/controllers/tasks_controller.rb w/app/controllers/tasks_controller.rb
index 062b1fa..431de4c 100644
--- i/app/controllers/tasks_controller.rb
+++ w/app/controllers/tasks_controller.rb
@@ -4,7 +4,7 @@
 class TasksController < ApplicationController
   def index
     @task = Task.new
-    @tasks = Task.all
+    @tasks = Task.all.order(created_at: :desc)
   end
 
   def search
diff --git i/app/views/tasks/index.html.erb w/app/views/tasks/index.html.erb
index 50263cb..5e1c0d8 100644
--- i/app/views/tasks/index.html.erb
+++ w/app/views/tasks/index.html.erb
@@ -1,9 +1,26 @@
 <div class="grid-x">
   <div class="cell small-12 medium-6 medium-offset-3 large-8 large-offset-2">
     <%= form_for @task do |f| %>
-      <%= f.text_field :subject, placehoder: t('task.new.placeholder'), required: true %>
-      <%= f.submit %>
+      <div class="input-group">
+        <%= f.text_field :subject,
+          placehoder: t('task.new.placeholder'),
+          required: true,
+          class: 'input-group-field',
+          label: false,
+          placeholder: 'Co teraz robisz?',
+          id: 'query' %>
+        <div class="input-group-button">
+          <%= f.submit class: 'button' %>
+        </div>
+      </div>
     <% end %>
+
+    <div id="suggestions">
+      <table>
+        <tbody>
+        </tbody>
+      </table>
+    </div>
   </div>
 </div>
 
@@ -15,6 +32,9 @@
           <td>
             <%= task.subject %>
           </td>
+          <td>
+            <%= l task.created_at %>
+          </td>
         </tr>
       <% end %>
     </table>
